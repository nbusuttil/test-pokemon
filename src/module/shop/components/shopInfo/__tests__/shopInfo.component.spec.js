import React from 'react';
import { shallow } from 'enzyme';

import ShopInfo from '../shopInfo.component';

describe('ShopInfo component', () => {
  let props;

  beforeEach(() => {
    props = {
      loadCart: jest.fn(),
      openModal: jest.fn(),
      removeItem: jest.fn(),
      addQuantity: jest.fn(),
      removeQuantity: jest.fn(),
      shop: {},
      cartSubtotal: 7
    };
  });
  const getWrapper = () => shallow(<ShopInfo {...props} />);

  it('render without crashing', () => {
    expect(getWrapper()).toMatchSnapshot();
  });

  it('render with item', () => {
    props.shop = {
      17: {
        id: 17,
        name: 'pidgeotto',
        price: 6,
        quantity: 1,
        url: 'https://pokeapi.co/api/v2/pokemon/17/'
      }
    };
    expect(getWrapper()).toMatchSnapshot();
  });
});
