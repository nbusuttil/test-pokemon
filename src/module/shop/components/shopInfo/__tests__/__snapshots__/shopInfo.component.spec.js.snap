// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopInfo component render with item 1`] = `
<div
  className="shopInfo"
>
  <div
    className="shopInfo__container"
  >
    <div
      className="shopInfo__container-card"
      key="17"
    >
      <Card
        action={[MockFunction]}
        addQuantity={[MockFunction]}
        isShop={true}
        modifier={true}
        openModal={[MockFunction]}
        pokemon={
          Object {
            "id": 17,
            "name": "pidgeotto",
            "price": 6,
            "quantity": 1,
            "url": "https://pokeapi.co/api/v2/pokemon/17/",
          }
        }
        removeQuantity={[MockFunction]}
        titleAction="Remove"
        titleAdd="+"
        titleRemove="-"
      />
    </div>
    <div
      className="shopInfo__container-actions"
    >
      <p>
        Cart Subtotal: 
        7
        â‚¬
      </p>
      <button>
        Buy
      </button>
    </div>
  </div>
</div>
`;

exports[`ShopInfo component render without crashing 1`] = `
<div
  className="shopInfo"
>
  <p
    className="shopInfo__empty"
  >
    Your Shopping Cart is empty.
  </p>
</div>
`;
